src/app/portlets/dot-edit-content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.ts

```diff
@@ -0,0 +1,108 @@
+
+    constructor(
+        private dotDevicesService: DotDevicesService,
+        private dotLanguagesService: DotLanguagesService,
+        private dotPersonasService: DotPersonasService,
+        private dotViewAsService: DotViewAsService,
+        private dotMessageService: DotMessageService
+    ) {}
+
+    ngOnInit() {
+        this.viewAsConfig = { language: null, persona: null };
+        Observable.forkJoin(
+            this.dotPersonasService.get(),
+            this.dotLanguagesService.get(),
+            this.dotDevicesService.get(),
+            this.dotMessageService.getMessages(['modes.persona.no.persona'])
+        ).subscribe(response => {
+            this.personasOptions = response[0].map((persona: DotPersona) => this.getPersonaFieldOption(persona));
+            this.personasOptions.unshift({
+                label: this.dotMessageService.get('modes.persona.no.persona'),
+                value: { identifier: '0' }
+            });
+            this.languagesOptions = response[1].map((language: DotLanguage) => this.getLanguageFieldOption(language));
+            this.devicesOptions = response[2].map((device: DotDevice) => this.getDeviceFieldOption(device));
+            this.setInitialConfiguration();
+        });
+    }
+
+    /**
+     * Track changes in dot-language and dot-persona.
+     */
+    changeConfiguration() {
+        this.dotViewAsService.selected = this.viewAsConfig;
+        this.changeViewAs.emit(this.dotViewAsService.selected);
+    }
+
+    /**
+     * Track changes in the dot-device.
+     */
+    changeDeviceConfiguration() {
+        this.dotViewAsService.selected = this.viewAsConfig;
+        this.changeDevice.emit(this.viewAsConfig.device);
+    }
+
+    private setInitialConfiguration() {
+        this.viewAsConfig = this.dotViewAsService.selected || {
+            persona: this.personasOptions[0].value,
+            device: this.devicesOptions[0].value,
+            language: this.languagesOptions[0].value
+        };
+        this.setPersonaDeviceInitialValue();
+        this.changeDevice.emit(this.viewAsConfig.device);
+    }
+
+    private setPersonaDeviceInitialValue() {
+        // Check apart because RenderHHTML service don't bring device & not always Persona
+        this.viewAsConfig.persona = this.viewAsConfig.persona || this.personasOptions[0].value;
+        this.viewAsConfig.device = this.viewAsConfig.device || this.devicesOptions[0].value;
+    }
+
+    private getPersonaFieldOption(persona: DotPersona): SelectItem {
+        return {
+            label: persona.name,
+            value: persona
+        };
+    }
+
+    private getLanguageFieldOption(language: DotLanguage): SelectItem {
+        return {
+            label: language.language,
+            value: language
+        };
+    }
+
+    private getDeviceFieldOption(device: DotDevice): SelectItem {
+        return {
+            label: device.label,
+            value: device
+        };
+    }
```

Comment:
I think  we dont need any of these three methods If we use optionLabel instead of options into the p-dropdown component